Installation issue: serf on A64FX, Fedora 33

[jw@cn05 jw]$ spack/bin/spack install serf
[+] /data/jw/spack/opt/spack/linux-centos8-a64fx/gcc-8.3.1/pkgconf-1.7.3-kdzlczv6oqysvfvd6kwnpkjjblqoiame
[+] /data/jw/spack/opt/spack/linux-centos8-a64fx/gcc-8.3.1/libbsd-0.10.0-uyaxcptvjncrbqd4ll4fgfw3upb2tk6g
[+] /data/jw/spack/opt/spack/linux-centos8-a64fx/gcc-8.3.1/libiconv-1.16-pp6ex45xatkqm57evnnymts2e3jkjtgc
[+] /data/jw/spack/opt/spack/linux-centos8-a64fx/gcc-8.3.1/berkeley-db-18.1.40-xhqs6cvlnggxtzkysk2fr5rdmzkbar3h
[+] /data/jw/spack/opt/spack/linux-centos8-a64fx/gcc-8.3.1/zlib-1.2.11-nssknf4ybspjgs5ip3g52h6gdsg7xvw4
[+] /data/jw/spack/opt/spack/linux-centos8-a64fx/gcc-8.3.1/xz-5.2.5-jtmowx7dpqrewoquz7adz5ljn6av2wcm
[+] /data/jw/spack/opt/spack/linux-centos8-a64fx/gcc-8.3.1/libffi-3.3-irtrfdvukdvrcxy3lehlgll7s5g7blm4
[+] /data/jw/spack/opt/spack/linux-centos8-a64fx/gcc-8.3.1/util-linux-uuid-2.36-p2ovzsijah54no7tv6cieqkkygtmfzho
[+] /data/jw/spack/opt/spack/linux-centos8-a64fx/gcc-8.3.1/ncurses-6.2-vsgvk3ktw3v5pvbdcdnqjynfjwf26lx3
[+] /data/jw/spack/opt/spack/linux-centos8-a64fx/gcc-8.3.1/expat-2.2.10-asbrkcc4enoin5hx72y2wwhse5l7wlyw
[+] /data/jw/spack/opt/spack/linux-centos8-a64fx/gcc-8.3.1/diffutils-3.7-p5u2pdhngtumslqt5gjzod2p5cjiqilf
[+] /data/jw/spack/opt/spack/linux-centos8-a64fx/gcc-8.3.1/tar-1.32-25ss6x3ehcyflvuqyo3v4ajgmb7kzmii
[+] /data/jw/spack/opt/spack/linux-centos8-a64fx/gcc-8.3.1/libxml2-2.9.10-amicqitgaop2jx3kahmques5ck4p3ngp
[+] /data/jw/spack/opt/spack/linux-centos8-a64fx/gcc-8.3.1/apr-1.7.0-lhpfpk5b4ozw5heictysynb2byphrbrf
[+] /data/jw/spack/opt/spack/linux-centos8-a64fx/gcc-8.3.1/readline-8.0-2bgw5veq6zdka6jq2rwqtitmt3rw3l3i
[+] /data/jw/spack/opt/spack/linux-centos8-a64fx/gcc-8.3.1/bzip2-1.0.8-tk2t636zh35aaclggn2pk76f6v5qw2js
[+] /data/jw/spack/opt/spack/linux-centos8-a64fx/gcc-8.3.1/gdbm-1.18.1-2dz44zlqyllstopcd42uorqaea6lne7p
[+] /data/jw/spack/opt/spack/linux-centos8-a64fx/gcc-8.3.1/sqlite-3.34.0-yn5c6am6zpkqd3kjfbhrtiwjyevpoj5t
[+] /data/jw/spack/opt/spack/linux-centos8-a64fx/gcc-8.3.1/gettext-0.21-j2w7r34kq73eirvbgzwvcmueiy2dyoso
[+] /data/jw/spack/opt/spack/linux-centos8-a64fx/gcc-8.3.1/perl-5.32.0-frdefhbg5p3mvsvixidnnh4sp6n6n7xf
[+] /data/jw/spack/opt/spack/linux-centos8-a64fx/gcc-8.3.1/openssl-1.1.1i-y3kre56snhsybjhcktyywmbvmfqtmwi6
[+] /data/jw/spack/opt/spack/linux-centos8-a64fx/gcc-8.3.1/python-3.8.7-uduadrnq3zce7h5jm332dcwoy5wumtdm
[+] /data/jw/spack/opt/spack/linux-centos8-a64fx/gcc-8.3.1/apr-util-1.6.1-2czpefrs3awuqz5et6g5vbx5dlxg6aws
[+] /data/jw/spack/opt/spack/linux-centos8-a64fx/gcc-8.3.1/py-setuptools-50.3.2-xwri6o6itpvigwdnnf5tnp6ipxsb3v37
[+] /data/jw/spack/opt/spack/linux-centos8-a64fx/gcc-8.3.1/scons-3.1.2-mwvskpu23abo3jaamjrjo6kvupmashxd
==> Installing serf-1.3.9-2gxtclmolm7b2fn7r4jo3r57gsjpdkle
==> No binary for serf-1.3.9-2gxtclmolm7b2fn7r4jo3r57gsjpdkle found: installing from source
==> Using cached archive: /data/jw/spack/var/spack/cache/_source-cache/archive/54/549c2d21c577a8a9c0450facb5cca809f26591f048e466552240947bdf7a87cc.tar.bz2
==> serf: Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    'scons' '-j16' 'PREFIX=/data/jw/spack/opt/spack/linux-centos8-a64fx/gcc-8.3.1/serf-1.3.9-2gxtclmolm7b2fn7r4jo3r57gsjpdkle' 'APR=/data/jw/spack/opt/spack/linux-centos8-a64fx/gcc-8.3.1/apr-1.7.0-lhpfpk5b4ozw5heictysynb2byphrbrf' 'APU=/data/jw/spack/opt/spack/linux-centos8-a64fx/gcc-8.3.1/apr-util-1.6.1-2czpefrs3awuqz5et6g5vbx5dlxg6aws' 'OPENSSL=/data/jw/spack/opt/spack/linux-centos8-a64fx/gcc-8.3.1/openssl-1.1.1i-y3kre56snhsybjhcktyywmbvmfqtmwi6' 'ZLIB=/data/jw/spack/opt/spack/linux-centos8-a64fx/gcc-8.3.1/zlib-1.2.11-nssknf4ybspjgs5ip3g52h6gdsg7xvw4' 'LINKFLAGS=-L/data/jw/spack/opt/spack/linux-centos8-a64fx/gcc-8.3.1/zlib-1.2.11-nssknf4ybspjgs5ip3g52h6gdsg7xvw4/lib -Wl,-rpath,/data/jw/spack/opt/spack/linux-centos8-a64fx/gcc-8.3.1/zlib-1.2.11-nssknf4ybspjgs5ip3g52h6gdsg7xvw4/lib' 'CPPFLAGS=-I/data/jw/spack/opt/spack/linux-centos8-a64fx/gcc-8.3.1/zlib-1.2.11-nssknf4ybspjgs5ip3g52h6gdsg7xvw4/include' 'DEBUG=no' 'CC=/usr/bin/gcc'

2 errors found in build log:
     83    /usr/include/string.h:384:15: note: expected 'const char *' but argument is of type 'unsigned char *'
     84     extern size_t strlen (const char *__s)
     85                   ^~~~~~
     86    ar rc libserf-1.a context.o incoming.o outgoing.o ssltunnel.o buckets/aggregate_buckets.o buckets/allocator.
           o buckets/barrier_buckets.o buckets/buckets.o buckets/bwtp_buckets.o buckets/chunk_buckets.o buckets/dechunk
           _buckets.o buckets/deflate_buckets.o buckets/file_buckets.o buckets/headers_buckets.o buckets/iovec_buckets.
           o buckets/limit_buckets.o buckets/mmap_buckets.o buckets/request_buckets.o buckets/response_body_buckets.o b
           uckets/response_buckets.o buckets/simple_buckets.o buckets/socket_buckets.o buckets/ssl_buckets.o auth/auth.
           o auth/auth_basic.o auth/auth_digest.o auth/auth_spnego.o auth/auth_spnego_gss.o auth/auth_spnego_sspi.o
     87    ranlib libserf-1.a
     88    /usr/bin/gcc -o libserf-1.so.1.3.0 -L/data/jw/spack/opt/spack/linux-centos8-a64fx/gcc-8.3.1/zlib-1.2.11-nssk
           nf4ybspjgs5ip3g52h6gdsg7xvw4/lib -Wl,-rpath,/data/jw/spack/opt/spack/linux-centos8-a64fx/gcc-8.3.1/zlib-1.2.
           11-nssknf4ybspjgs5ip3g52h6gdsg7xvw4/lib -pthread -shared -Wl,-Bsymbolic -Wl,-soname=libserf-1.so.1 -Wl,-rpat
           h=/data/jw/spack/opt/spack/linux-centos8-a64fx/gcc-8.3.1/serf-1.3.9-2gxtclmolm7b2fn7r4jo3r57gsjpdkle/lib -Wl
           ,-rpath=/data/jw/spack/opt/spack/linux-centos8-a64fx/gcc-8.3.1/apr-1.7.0-lhpfpk5b4ozw5heictysynb2byphrbrf/li
           b -Wl,-rpath=/data/jw/spack/opt/spack/linux-centos8-a64fx/gcc-8.3.1/openssl-1.1.1i-y3kre56snhsybjhcktyywmbvm
           fqtmwi6/lib -Wl,-rpath=/data/jw/spack/opt/spack/linux-centos8-a64fx/gcc-8.3.1/expat-2.2.10-asbrkcc4enoin5hx7
           2y2wwhse5l7wlyw/lib -Wl,-rpath=/data/jw/spack/opt/spack/linux-centos8-a64fx/gcc-8.3.1/libiconv-1.16-pp6ex45x
           atkqm57evnnymts2e3jkjtgc/lib -Wl,-rpath=/data/jw/spack/opt/spack/linux-centos8-a64fx/gcc-8.3.1/apr-util-1.6.
           1-2czpefrs3awuqz5et6g5vbx5dlxg6aws/lib -Wl,-rpath=/data/jw/spack/opt/spack/linux-centos8-a64fx/gcc-8.3.1/ope
           nssl-1.1.1i-y3kre56snhsybjhcktyywmbvmfqtmwi6/lib context.os incoming.os outgoing.os ssltunnel.os buckets/agg
           regate_buckets.os buckets/allocator.os buckets/barrier_buckets.os buckets/buckets.os buckets/bwtp_buckets.os
            buckets/chunk_buckets.os buckets/dechunk_buckets.os buckets/deflate_buckets.os buckets/file_buckets.os buck
           ets/headers_buckets.os buckets/iovec_buckets.os buckets/limit_buckets.os buckets/mmap_buckets.os buckets/req
           uest_buckets.os buckets/response_body_buckets.os buckets/response_buckets.os buckets/simple_buckets.os bucke
           ts/socket_buckets.os buckets/ssl_buckets.os auth/auth.os auth/auth_basic.os auth/auth_digest.os auth/auth_sp
           nego.os auth/auth_spnego_gss.os auth/auth_spnego_sspi.os -L/data/jw/spack/opt/spack/linux-centos8-a64fx/gcc-
           8.3.1/apr-1.7.0-lhpfpk5b4ozw5heictysynb2byphrbrf/lib -L/data/jw/spack/opt/spack/linux-centos8-a64fx/gcc-8.3.
           1/openssl-1.1.1i-y3kre56snhsybjhcktyywmbvmfqtmwi6/lib -L/data/jw/spack/opt/spack/linux-centos8-a64fx/gcc-8.3
           .1/expat-2.2.10-asbrkcc4enoin5hx72y2wwhse5l7wlyw/lib -L/data/jw/spack/opt/spack/linux-centos8-a64fx/gcc-8.3.
           1/libiconv-1.16-pp6ex45xatkqm57evnnymts2e3jkjtgc/lib -L/data/jw/spack/opt/spack/linux-centos8-a64fx/gcc-8.3.
           1/apr-util-1.6.1-2czpefrs3awuqz5et6g5vbx5dlxg6aws/lib -L/data/jw/spack/opt/spack/linux-centos8-a64fx/gcc-8.3
           .1/openssl-1.1.1i-y3kre56snhsybjhcktyywmbvmfqtmwi6/lib -lssl -lcrypto -lz -lapr-1 -luuid -lrt -lcrypt -lpthr
           ead -ldl -laprutil-1 -lexpat -liconv
  >> 89    /bin/ld: cannot find -luuid
  >> 90    collect2: error: ld returned 1 exit status
     91    scons: *** [libserf-1.so.1.3.0] Error 1
     92    scons: building terminated because of errors.

See build log for details:
  /tmp/jw/spack-stage/spack-stage-serf-1.3.9-2gxtclmolm7b2fn7r4jo3r57gsjpdkle/spack-build-out.txt

[jw@cn05 jw]$ spack/bin/spack debug report
* **Spack:** 0.16.0-948-9d1d37c432
* **Python:** 3.6.8
* **Platform:** linux-centos8-a64fx
* **Concretizer:** original
[jw@cn05 jw]$

Hi @adamjstewart ,

Thank you for the suggestions and thank you @LutzWeischerFujitsu for opening this issue.

I am posting this message because I'd like to help make this package installation work with *spack*.

In my case, I am finding this issue when trying to get *sphinx* (which needs, according to https://spack.readthedocs.io/en/latest/contribution_guide.html, *subversion*) with the configuration below (and gcc@11.1.0 installed via spack):

* **Spack:** 0.16.2-3116-0d73fd2b11
* **Python:** 3.8.2
* **Platform:** linux-ubuntu20.04-skylake
* **Concretizer:** original

And I tested adding the dependency, using the command `spack edit serf` and each of the various lines below:
```
    depends_on('uuid')
    #depends_on('libuuid', type='build', when='@1.3.9')
    #depends_on('util-linux-uuid', type='build', when='@1.3.9')
    #depends_on('uuid', type='build', when='@1.3.9')
```

But I still seem to find the same issue:
```
$ spack install serf
(...)
[+] /mnt/d/Diana/opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-11.1.0/util-linux-uuid-2.36.2-xrah6np576rpy267tomvw3443utiuhvd
(...)
==> Installing serf-1.3.9-jtrumf6sprakw3yccaxbr6qnfs54ef5b
==> No binary for serf-1.3.9-jtrumf6sprakw3yccaxbr6qnfs54ef5b found: installing from source
==> Using cached archive: /mnt/d/Diana/opt/spack/var/spack/cache/_source-cache/archive/54/549c2d21c577a8a9c0450facb5cca809f26591f048e466552240947bdf7a87cc.tar.bz2
==> Applied patch /mnt/d/Diana/opt/spack/var/spack/repos/builtin/packages/serf/py3syntax.patch
==> serf: Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    'scons' '-j4' 'PREFIX=/mnt/d/Diana/opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-11.1.0/serf-1.3.9-jtrumf6sprakw3yccaxbr6qnfs54ef5b' 'APR=/mnt/d/Diana/opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-11.1.0/apr-1.7.0-pn34wwu7jsrc67o54mjxyfolbzpudezk' 'APU=/mnt/d/Diana/opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-11.1.0/apr-util-1.6.1-w4jvfslfm53ioathip4ub5vsbozwcdpe' 'OPENSSL=/mnt/d/Diana/opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-11.1.0/openssl-1.1.1k-svl2fsiyre3jtm2gwzizhaahnmr7nnoj' 'ZLIB=/mnt/d/Diana/opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-11.1.0/zlib-1.2.11-7ufhpo6xixdtl6c7mndowh2phba2qavm' 'LINKFLAGS=-L/mnt/d/Diana/opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-11.1.0/zlib-1.2.11-7ufhpo6xixdtl6c7mndowh2phba2qavm/lib -Wl,-rpath,/mnt/d/Diana/opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-11.1.0/zlib-1.2.11-7ufhpo6xixdtl6c7mndowh2phba2qavm/lib' 'CPPFLAGS=-I/mnt/d/Diana/opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-11.1.0/zlib-1.2.11-7ufhpo6xixdtl6c7mndowh2phba2qavm/include' 'DEBUG=no' 'CC=/mnt/d/Diana/opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.3.0/gcc-11.1.0-plmek74g4nwxu4iiexwepnmgm2hckit7/bin/gcc'

2 errors found in build log:
     115          |                       ~~~~~~~~~~~~^~~
     116    /mnt/d/Diana/opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.3.0/gcc-11.1.0-plmek74g4nwxu4iiexwepnmgm2h
            ckit7/bin/gcc -o auth/auth_spnego.os -c -std=c89 -Wdeclaration-after-statement -Wmissing-prototypes -Wall -
            g -O2 -pthread -fPIC -I/mnt/d/Diana/opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-11.1.0/zlib-1.2.11-7u
            fhpo6xixdtl6c7mndowh2phba2qavm/include -DNDEBUG -DLINUX -D_REENTRANT -D_GNU_SOURCE -I. -I/mnt/d/Diana/opt/s
            pack/opt/spack/linux-ubuntu20.04-skylake/gcc-11.1.0/apr-1.7.0-pn34wwu7jsrc67o54mjxyfolbzpudezk/include/apr-
            1 -I/mnt/d/Diana/opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-11.1.0/apr-util-1.6.1-w4jvfslfm53ioathip
            4ub5vsbozwcdpe/include/apr-1 -I/mnt/d/Diana/opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-11.1.0/openss
            l-1.1.1k-svl2fsiyre3jtm2gwzizhaahnmr7nnoj/include -I/mnt/d/Diana/opt/spack/opt/spack/linux-ubuntu20.04-skyl
            ake/gcc-11.1.0/expat-2.3.0-gfyyffd4wn5kliqkyn7imwttflyxc5pn/include -I/mnt/d/Diana/opt/spack/opt/spack/linu
            x-ubuntu20.04-skylake/gcc-11.1.0/libiconv-1.16-qvc5pzwuiqmzskehjhid5lcmsy63rfkq/include -I/mnt/d/Diana/opt/
            spack/opt/spack/linux-ubuntu20.04-skylake/gcc-11.1.0/openssl-1.1.1k-svl2fsiyre3jtm2gwzizhaahnmr7nnoj/includ
            e auth/auth_spnego.c
     117    /mnt/d/Diana/opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.3.0/gcc-11.1.0-plmek74g4nwxu4iiexwepnmgm2h
            ckit7/bin/gcc -o auth/auth_spnego_gss.os -c -std=c89 -Wdeclaration-after-statement -Wmissing-prototypes -Wa
            ll -g -O2 -pthread -fPIC -I/mnt/d/Diana/opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-11.1.0/zlib-1.2.1
            1-7ufhpo6xixdtl6c7mndowh2phba2qavm/include -DNDEBUG -DLINUX -D_REENTRANT -D_GNU_SOURCE -I. -I/mnt/d/Diana/o
            pt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-11.1.0/apr-1.7.0-pn34wwu7jsrc67o54mjxyfolbzpudezk/include/
            apr-1 -I/mnt/d/Diana/opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-11.1.0/apr-util-1.6.1-w4jvfslfm53ioa
            thip4ub5vsbozwcdpe/include/apr-1 -I/mnt/d/Diana/opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-11.1.0/op
            enssl-1.1.1k-svl2fsiyre3jtm2gwzizhaahnmr7nnoj/include -I/mnt/d/Diana/opt/spack/opt/spack/linux-ubuntu20.04-
            skylake/gcc-11.1.0/expat-2.3.0-gfyyffd4wn5kliqkyn7imwttflyxc5pn/include -I/mnt/d/Diana/opt/spack/opt/spack/
            linux-ubuntu20.04-skylake/gcc-11.1.0/libiconv-1.16-qvc5pzwuiqmzskehjhid5lcmsy63rfkq/include -I/mnt/d/Diana/
            opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-11.1.0/openssl-1.1.1k-svl2fsiyre3jtm2gwzizhaahnmr7nnoj/in
            clude auth/auth_spnego_gss.c
     118    /mnt/d/Diana/opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.3.0/gcc-11.1.0-plmek74g4nwxu4iiexwepnmgm2h
            ckit7/bin/gcc -o auth/auth_spnego_sspi.os -c -std=c89 -Wdeclaration-after-statement -Wmissing-prototypes -W
            all -g -O2 -pthread -fPIC -I/mnt/d/Diana/opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-11.1.0/zlib-1.2.
            11-7ufhpo6xixdtl6c7mndowh2phba2qavm/include -DNDEBUG -DLINUX -D_REENTRANT -D_GNU_SOURCE -I. -I/mnt/d/Diana/
            opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-11.1.0/apr-1.7.0-pn34wwu7jsrc67o54mjxyfolbzpudezk/include
            /apr-1 -I/mnt/d/Diana/opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-11.1.0/apr-util-1.6.1-w4jvfslfm53io
            athip4ub5vsbozwcdpe/include/apr-1 -I/mnt/d/Diana/opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-11.1.0/o
            penssl-1.1.1k-svl2fsiyre3jtm2gwzizhaahnmr7nnoj/include -I/mnt/d/Diana/opt/spack/opt/spack/linux-ubuntu20.04
            -skylake/gcc-11.1.0/expat-2.3.0-gfyyffd4wn5kliqkyn7imwttflyxc5pn/include -I/mnt/d/Diana/opt/spack/opt/spack
            /linux-ubuntu20.04-skylake/gcc-11.1.0/libiconv-1.16-qvc5pzwuiqmzskehjhid5lcmsy63rfkq/include -I/mnt/d/Diana
            /opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-11.1.0/openssl-1.1.1k-svl2fsiyre3jtm2gwzizhaahnmr7nnoj/i
            nclude auth/auth_spnego_sspi.c
     119    Creating 'serf-1.pc'
     120    /mnt/d/Diana/opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-9.3.0/gcc-11.1.0-plmek74g4nwxu4iiexwepnmgm2h
            ckit7/bin/gcc -o libserf-1.so.1.3.0 -L/mnt/d/Diana/opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-11.1.0
            /zlib-1.2.11-7ufhpo6xixdtl6c7mndowh2phba2qavm/lib -Wl,-rpath,/mnt/d/Diana/opt/spack/opt/spack/linux-ubuntu2
            0.04-skylake/gcc-11.1.0/zlib-1.2.11-7ufhpo6xixdtl6c7mndowh2phba2qavm/lib -pthread -shared -Wl,-Bsymbolic -W
            l,-soname=libserf-1.so.1 -Wl,-rpath=/mnt/d/Diana/opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-11.1.0/s
            erf-1.3.9-jtrumf6sprakw3yccaxbr6qnfs54ef5b/lib -Wl,-rpath=/mnt/d/Diana/opt/spack/opt/spack/linux-ubuntu20.0
            4-skylake/gcc-11.1.0/apr-1.7.0-pn34wwu7jsrc67o54mjxyfolbzpudezk/lib -Wl,-rpath=/mnt/d/Diana/opt/spack/opt/s
            pack/linux-ubuntu20.04-skylake/gcc-11.1.0/openssl-1.1.1k-svl2fsiyre3jtm2gwzizhaahnmr7nnoj/lib -Wl,-rpath=/m
            nt/d/Diana/opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-11.1.0/expat-2.3.0-gfyyffd4wn5kliqkyn7imwttfly
            xc5pn/lib -Wl,-rpath=/mnt/d/Diana/opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-11.1.0/libiconv-1.16-qv
            c5pzwuiqmzskehjhid5lcmsy63rfkq/lib -Wl,-rpath=/mnt/d/Diana/opt/spack/opt/spack/linux-ubuntu20.04-skylake/gc
            c-11.1.0/apr-util-1.6.1-w4jvfslfm53ioathip4ub5vsbozwcdpe/lib -Wl,-rpath=/mnt/d/Diana/opt/spack/opt/spack/li
            nux-ubuntu20.04-skylake/gcc-11.1.0/openssl-1.1.1k-svl2fsiyre3jtm2gwzizhaahnmr7nnoj/lib context.os incoming.
            os outgoing.os ssltunnel.os buckets/aggregate_buckets.os buckets/allocator.os buckets/barrier_buckets.os bu
            ckets/buckets.os buckets/bwtp_buckets.os buckets/chunk_buckets.os buckets/dechunk_buckets.os buckets/deflat
            e_buckets.os buckets/file_buckets.os buckets/headers_buckets.os buckets/iovec_buckets.os buckets/limit_buck
            ets.os buckets/mmap_buckets.os buckets/request_buckets.os buckets/response_body_buckets.os buckets/response
            _buckets.os buckets/simple_buckets.os buckets/socket_buckets.os buckets/ssl_buckets.os auth/auth.os auth/au
            th_basic.os auth/auth_digest.os auth/auth_spnego.os auth/auth_spnego_gss.os auth/auth_spnego_sspi.os -L/mnt
            /d/Diana/opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-11.1.0/apr-1.7.0-pn34wwu7jsrc67o54mjxyfolbzpudez
            k/lib -L/mnt/d/Diana/opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-11.1.0/openssl-1.1.1k-svl2fsiyre3jtm
            2gwzizhaahnmr7nnoj/lib -L/mnt/d/Diana/opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-11.1.0/expat-2.3.0-
            gfyyffd4wn5kliqkyn7imwttflyxc5pn/lib -L/mnt/d/Diana/opt/spack/opt/spack/linux-ubuntu20.04-skylake/gcc-11.1.
            0/libiconv-1.16-qvc5pzwuiqmzskehjhid5lcmsy63rfkq/lib -L/mnt/d/Diana/opt/spack/opt/spack/linux-ubuntu20.04-s
            kylake/gcc-11.1.0/apr-util-1.6.1-w4jvfslfm53ioathip4ub5vsbozwcdpe/lib -L/mnt/d/Diana/opt/spack/opt/spack/li
            nux-ubuntu20.04-skylake/gcc-11.1.0/openssl-1.1.1k-svl2fsiyre3jtm2gwzizhaahnmr7nnoj/lib -lssl -lcrypto -lz -
            lapr-1 -luuid -lrt -lcrypt -lpthread -ldl -laprutil-1 -lexpat -liconv
  >> 121    /bin/ld: cannot find -luuid
  >> 122    collect2: error: ld returned 1 exit status
     123    scons: *** [libserf-1.so.1.3.0] Error 1
     124    scons: building terminated because of errors.
```

So I tried adding the line `link_flags.extend(["-luuid"])` both before the current one of `args.append('LINKFLAGS=' + ' '.join(link_flags))` and alone before the `if` version ... (so it would affect any versions).

But still got the same error:
```
(...)
  >> 121    /bin/ld: cannot find -luuid
  >> 123    collect2: error: ld returned 1 exit status
```

The same happened when I tried using:
```
        if '@:1.3.9' in self.spec:
            uuid_spec = self.spec['uuid']
            link_flags_uuid = [uuid_spec.libs.search_flags]
            link_flags_uuid.extend(["-luuid"])
            args.append('LINKFLAGS=' + ' '.join(link_flags_uuid))
            args.append('CPPFLAGS=' + uuid_spec.headers.cpp_flags)
```

I wonder if you have other suggestion that I can try.
I can open a PR to fix this, if you prefer.

(I'm trying the work around of running `spack install subversion ~serf`)

Thanks

--- a/var/spack/repos/builtin/packages/apr-util/package.py
+++ b/var/spack/repos/builtin/packages/apr-util/package.py
@@ -32,6 +32,14 @@ class AprUtil(AutotoolsPackage):
     depends_on('sqlite', when='+sqlite')
     depends_on('unixodbc', when='+odbc')
 
+    @property
+    def libs(self):
+        return find_libraries(
+            ['libaprutil-{0}'.format(self.version.up_to(1))],
+            root=self.prefix,
+            recursive=True,
+        )
+
     def configure_args(self):
         spec = self.spec
 

--- a/var/spack/repos/builtin/packages/apr/package.py
+++ b/var/spack/repos/builtin/packages/apr/package.py
@@ -19,3 +19,11 @@ class Apr(AutotoolsPackage):
     patch('missing_includes.patch', when='@1.7.0')
 
     depends_on('uuid', type='link')
+
+    @property
+    def libs(self):
+        return find_libraries(
+            ['libapr-{0}'.format(self.version.up_to(1))],
+            root=self.prefix,
+            recursive=True,
+        )

--- a/var/spack/repos/builtin/packages/serf/package.py
+++ b/var/spack/repos/builtin/packages/serf/package.py
@@ -25,43 +25,42 @@ class Serf(SConsPackage):
     depends_on('apr-util')
     depends_on('openssl')
     depends_on('zlib')
+    depends_on('uuid')
 
     patch('py3syntax.patch')
 
     def build_args(self, spec, prefix):
-        args = [
-            'PREFIX={0}'.format(prefix),
-            'APR={0}'.format(spec['apr'].prefix),
-            'APU={0}'.format(spec['apr-util'].prefix),
-            'OPENSSL={0}'.format(spec['openssl'].prefix),
-            'ZLIB={0}'.format(spec['zlib'].prefix),
-        ]
-
-        # ZLIB variable is ignored on non-Windows platforms before and
-        # including the version 1.3.9:
-        # https://www.mail-archive.com/dev@serf.apache.org/msg01359.html
-        # The issue is fixed in the trunk. Hopefully, the next stable version
-        # will work properly.
-        if '@:1.3.9' in self.spec:
-            zlib_spec = self.spec['zlib']
-            link_flags = [zlib_spec.libs.search_flags]
-            link_flags.extend([self.compiler.cc_rpath_arg + d
-                               for d in zlib_spec.libs.directories])
-            args.append('LINKFLAGS=' + ' '.join(link_flags))
-            args.append('CPPFLAGS=' + zlib_spec.headers.cpp_flags)
-
-        if '+debug' in spec:
-            args.append('DEBUG=yes')
-        else:
-            args.append('DEBUG=no')
+        args = {
+            'PREFIX': prefix,
+            'APR': spec['apr'].prefix,
+            'APU': spec['apr-util'].prefix,
+            'OPENSSL': spec['openssl'].prefix,
+            'ZLIB': spec['zlib'].prefix,
+            'DEBUG': 'yes' if '+debug' in spec else 'no',
+        }
 
         # SCons doesn't pass Spack environment variables to the
         # execution environment. Therefore, we can't use Spack's compiler
         # wrappers. Use the actual compilers. SCons seems to RPATH things
         # on its own anyway.
-        args.append('CC={0}'.format(self.compiler.cc))
+        args['CC'] = self.compiler.cc
 
-        return args
+        # Old versions of serf ignore the ZLIB variable on non-Windows platforms.
+        # Also, there is no UUID variable to specify its installation location.
+        # Pass explicit link flags for both.
+        library_dirs = []
+        include_dirs = []
+        for dep in spec.dependencies(deptype='link'):
+            query = self.spec[dep.name]
+            library_dirs.extend(query.libs.directories)
+            include_dirs.extend(query.headers.directories)
+
+        rpath = self.compiler.cc_rpath_arg
+        args['LINKFLAGS'] = '-L' + ' -L'.join(library_dirs)
+        args['LINKFLAGS'] += ' ' + rpath + (' ' + rpath).join(library_dirs)
+        args['CPPFLAGS'] = '-I' + ' -I'.join(include_dirs)
+
+        return [key + '=' + value for key, value in args.items()]
 
     def build_test(self):
         # FIXME: Several test failures:

